tests/tick.test.mjs

```diff
@@ -657,4 +662,156 @@ describe('Tick', async () => {
+    const firstEpochMidTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: firstEpochMidTimestamp,
+      },
+      buyRecordMemory,
+    );
+    const feeDuringFirstEpoch = await getBaseRegistrationFee({
+      memory: firstEpochMidTick.Memory,
+      timestamp: firstEpochMidTimestamp + 1,
+    });
+
+    assert.equal(feeDuringFirstEpoch, 600_000_000);
+    const firstEpochDemandFactorResult = await getDemandFactor({
+      memory: firstEpochMidTick.Memory,
+      timestamp: firstEpochMidTimestamp + 1,
+    });
+    assert.equal(firstEpochDemandFactorResult, 1);
+
+    // Tick to the end of the first epoch
+    const firstEpochEndTimestamp =
+      genesisEpochStart + epochDurationMs + distributionDelayMs + 1;
+    const firstEpochEndTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: firstEpochEndTimestamp,
+      },
+      buyRecordMemory,
+    );
+    const feeAfterFirstEpochEnd = await getBaseRegistrationFee({
+      memory: firstEpochEndTick.Memory,
+      timestamp: firstEpochEndTimestamp + 1,
+    });
+
+    assert.equal(feeAfterFirstEpochEnd, 630_000_000);
+
+    const firstEpochEndDemandFactorResult = await getDemandFactor({
+      memory: firstEpochEndTick.Memory,
+      timestamp: firstEpochEndTimestamp + 1,
+    });
+    assert.equal(firstEpochEndDemandFactorResult, 1.0500000000000000444);
+  });
+
+  it('should reset to baseRegistrationFee when demandFactor is 0.5 for consecutive epochs', async () => {
+    const zeroEpochTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: genesisEpochStart,
+      },
+      startMemory,
+    );
+
+    const baseFeeAtZeroEpoch = await getBaseRegistrationFee({
+      memory: zeroEpochTick.Memory,
+      timestamp: genesisEpochStart,
+    });
+    assert.equal(baseFeeAtZeroEpoch, 600_000_000);
+
+    let tickMemory = zeroEpochTick.Memory;
+
+    // Tick to the epoch where demandFactor is 0.5
+    for (let i = 0; i <= 49; i++) {
+      const epochTimestamp = genesisEpochStart + (epochDurationMs + 1) * i;
+      const { Memory } = await handle(
+        {
+          Tags: [
+            { name: 'Action', value: 'Tick' },
+            { name: 'Timestamp', value: epochTimestamp.toString() },
+          ],
+          Timestamp: epochTimestamp,
+        },
+        tickMemory,
+      );
+      tickMemory = Memory;
+
+      if (i === 45) {
+        const demandFactor = await getDemandFactor({
+          memory: tickMemory,
+          timestamp: epochTimestamp,
+        });
+        assert.equal(demandFactor, 0.50655939255251769548);
+      }
+
+      if ([46, 47, 48].includes(i)) {
+        const demandFactor = await getDemandFactor({
+          memory: tickMemory,
+          timestamp: epochTimestamp,
+        });
+        assert.equal(demandFactor, 0.5);
+      }
+    }
+
+    const afterTimestamp = genesisEpochStart + (epochDurationMs + 1) * 50;
```

Comment:
Should this be:
```
genesisEpochStart + epochDurationMs * 50 + 1
```
?

Reply:
its implied with * 50 rather than 49. but setting the timestamps here don't seem to change the outcome --  the memory seems to be the only important variable for getDemandFactor and getBaseFees

Reply:
on the 49th tick since genesis we reset and return back to demandFactor 1